internal/channel_tags/database.go

```diff
@@ -0,0 +1,89 @@
+package channel_tags
+
+import (
+	"database/sql"
+	"github.com/jmoiron/sqlx"
+	"github.com/lncapital/torq/internal/channels"
+	"github.com/lncapital/torq/internal/corridors"
+	"github.com/lncapital/torq/internal/database"
+	"github.com/lncapital/torq/internal/tags"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"time"
+)
+
+func getChannelTag(db *sqlx.DB, channelTagId int) (channelTag, error) {
+	ct := channelTag{}
+	err := db.Get(&ct, "SELECT * FROM channel_tag WHERE channel_tag_id = $1;", channelTagId)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return channelTag{}, nil
+		}
+		return channelTag{}, errors.Wrap(err, database.SqlExecutionError)
+	}
+	return ct, nil
+}
+
+func addChannelTag(db *sqlx.DB, ct channelTag) (channelTag, error) {
+	ct.CreatedOn = time.Now().UTC()
+	// no updated_on because a channel tag is never update only insert/delete.
+	err := db.QueryRowx(`
+		INSERT INTO channel_tag (
+		  node_id,
+		  peer_id,
+		  channel_db_id,
+		  tag_id,
+		  created_on
+		) values ($1, $2, $3, $4, $5)
+		RETURNING channel_tag_id;`,
+		ct.NodeId, ct.PeerId, ct.ChannelId, ct.TagId, ct.CreatedOn).Scan(&ct.ChannelTagId)
+	if err != nil {
+		return channelTag{}, errors.Wrap(err, database.SqlExecutionError)
+	}
+
+	return ct, nil
+}
+
+func GenerateChannelTag(db *sqlx.DB) error {
+	channels, err := channels.GetAllChannels(db)
+	if err != nil {
+		return errors.Wrap(err, "Obtaining channels for channel tag generation.")
+	}
+	tags, err := tags.GetTags(db)
+	if err != nil {
+		return errors.Wrap(err, "Obtaining tags for channel tag generation.")
+	}
+	tx, err := db.Begin()
+	if err != nil {
+		return errors.Wrap(err, database.SqlBeginTransactionError)
+	}
+
+	_, err = tx.Exec("DELETE FROM channel_tag WHERE tag_origin_id=$1;", corridor)
+	if err != nil {
+		if rb := tx.Rollback(); rb != nil {
+			log.Error().Msgf("Attempted to abort: %v (%v)", err, rb)
```

Comment:
Logging here is the right thing to do but you don't need to include err because that's going to get bubbled up on the next line.

When using zerolog, you can give err to the err() function rather than formatting it in using Msgf. See my error handling document for an example.
```suggestion
			log.Error().Err(err).Msg("Unable to rollback transaction")
```